### PR TITLE
node+rpc: rpc environment should own it's creation

### PR DESCRIPTION
**Pull request by tychoish**
This is just in a queue behind/on top of #7572.

### CODE REVIEW

**Review by marbar3778 — COMMENTED** [review]
catching up on a bunch of prs, I love the work you are doing in the repo!! Thank you for doing the work that leads to less headaches

File: internal/rpc/core/env.go [hunk 1]
```diff
@@ -205,3 +212,95 @@ func (env *Environment) latestUncommittedHeight() int64 {
 	}
 	return env.BlockStore.Height() + 1
 }
+
+func (env *Environment) StartService(ctx context.Context, conf *config.Config) ([]net.Listener, error) {
```
Comment (cmwaters) [inline internal/rpc/core/env.go:216]: godoc would be nice

File: node/node.go [hunk 2]
```diff
@@ -433,6 +430,14 @@ func makeNode(
...
 
+	if cfg.Mode == config.ModeValidator {
+		pubKey, err := privValidator.GetPubKey(ctx)
+		if pubKey == nil || err != nil {
+			return nil, fmt.Errorf("can't get pubkey: %w", err)
+		}
+		node.rpcEnv.PubKey = pubKey
+	}
+
 	node.rpcEnv.P2PTransport = node
```
Comment (cmwaters) [inline node/node.go:441]: This is some weird shit I remember coming across that relates to querying `NodeInfo` but something we can do in a follow up issue perhaps

**Review by cmwaters — APPROVED** [review]
Great work!! One day that node package is going to be understandable

File: node/node.go (hunk 2 above)
Comment (tychoish) [reply node/node.go:441]: yeah, definitely.

File: internal/rpc/core/env.go (hunk 1 above)
Comment (tychoish) [reply internal/rpc/core/env.go:220]: done!

**Review by williambanfield — APPROVED** [review]

**Review by creachadair — APPROVED** [review]

**Pull request merged**